src/adhocracy_core/adhocracy_core/rest/schemas.py

```diff
@@ -488,3 +496,80 @@ def _add_arbitrary_filter_node(name, index: SDIndex, schema):
                               'data': {}},
              'response_body': {'content_type': '',
                                'path': ''}}}
+
+
+@colander.deferred
+def deferred_validate_password_reset_email(node: SchemaNode, kw: dict):
+    """Validate the email address of a password reset request.
+
+    If valid, the user object is added as 'user' to
+    `request.validated`.
+
+    :raise colander.Invalid: if no user with this email exists.
+    """
+    context = kw['context']
+    request = kw['request']
+
+    def validate_email(node, value):
+        locator = request.registry.getMultiAdapter((context, request),
+                                                   IUserLocator)
+        user = locator.get_user_by_email(value)
+        if user is None:
+            msg = 'No user exists with this email: {0}'.format(value)
+            raise colander.Invalid(node, msg)
```

Comment:
Yes, but this can already be done through `register` as well.

If we want to avoid that, this has a rather high impact on UX, i.e. allow to register, but maybe without success - you need to check your emails first. Same for "forget password" - we would then need to write "if you have an account, you'll now get an email".

We can improve this later if we have a good idea.